community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -235,7 +236,32 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
       indexSeekByPrefixRange(index, prefix)
     case range: InequalitySeekRange[Any] =>
       indexSeekByPrefixRange(index, range)
+    case range: PointDistanceRange[Any] =>
+      val distance: Double = range.distance match {
+        case n:NumberValue => n.doubleValue()
+        case n:Number => n.doubleValue()
+        case _ =>
+          // We can't compare against something that is not a number, so no rows will match
+          return Iterator.empty
+      }
+
+      val (from, to) = range.point match {
+        case p:PointValue if p.getCoordinateReferenceSystem == Cartesian =>
+          CartesianCalculator.boundingBox(p, distance)
```

Comment:
That happens in the leaf planner where we don't report the predicates as solved, so that the later planning steps are forced to plan a filter for those predicates. See my response to Andres request for a comment above. I suggested the following:

> The planned index seek will _almost_ satisfy the predicate, but with the possibility of some false positives. Since it reduces the cardinality to almost the level of the predicate, we can use the predicate to calculate cardinality, but not mark it as solved, since the planner will still need to solve it with a Filter.